orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Stage.java

```diff
@@ -338,21 +338,44 @@ public Task taskById(String taskId) {
     return tasks.stream().filter(it -> it.getId().equals(taskId)).findFirst().orElse(null);
   }
 
-  /** Gets all ancestor stages, including the current stage. */
+  /**
+   * Gets all ancestor stages, including the current stage.
+   *
+   * <p>Ancestors include: <br>
+   * - stages this stage depends on (via requisiteStageRefIds) <br>
+   * - parent stages of this stage <br>
+   * - synthetic stages that share a parent with this stage & occur prior to current stage <br>
+   */
   public List<Stage> ancestors() {
     if (execution != null) {
       Set<String> visited = Sets.newHashSetWithExpectedSize(execution.getStages().size());
-      return ImmutableList.<Stage>builder().add(this).addAll(ancestorsOnly(visited)).build();
+      return ImmutableList.<Stage>builder().add(this).addAll(ancestorsOnly(visited, false)).build();
     } else {
       return emptyList();
     }
   }
 
-  private List<Stage> ancestorsOnly(Set<String> visited) {
+  /**
+   * Gets all ancestor stages that are direct parents, including the current stage.
+   *
+   * <p>Ancestors include: <br>
+   * - parent stages of this stage <br>
+   * - synthetic stages that share a parent with this stage & occur prior to current stage
+   */
+  public List<Stage> directAncestors() {
+    if (execution != null) {
+      Set<String> visited = Sets.newHashSetWithExpectedSize(execution.getStages().size());
+      return ImmutableList.<Stage>builder().add(this).addAll(ancestorsOnly(visited, true)).build();
+    } else {
+      return emptyList();
+    }
+  }
+
+  private List<Stage> ancestorsOnly(Set<String> visited, boolean directParentOnly) {
```

Comment:
I think I can make sense of the `ancestors` and `directAncestors` method names, but I'm not sure what `ancestorsOnly` is supposed to mean 🤔

Reply:
yeah, this was here before... but yeah, i don't like it... in my past life I would call a method like this `getAncestorsImpl` for being the actual implementation but I think that's the convention in Java?...

maybe just rename it to `ancestors` or `getAncestors` and have the fact that it's `private` be the indication that it's the "worker"?

Reply:
up to you, I wouldn't be opposed to `getAncestorsImpl`